DirectX11/HackerDXGI.cpp

```diff
@@ -167,6 +167,104 @@ HackerDXGISwapChain::HackerDXGISwapChain(IDXGISwapChain *pSwapChain, HackerDevic
 	}
 }
 
+HackerDXGISwapChain::HackerDXGISwapChain(IDXGISwapChain *pSwapChain, HackerDevice *pDevice, HackerContext *pContext, const DXGI_SWAP_CHAIN_DESC* fake_swap_chain_desc, UINT new_width, UINT new_height)
+	: HackerDXGIDeviceSubObject(pSwapChain), mFakeBackBuffer(nullptr), mFakeSwapChain(nullptr), mWidth(0), mHeight(0)
+{
+	mOrigSwapChain = pSwapChain;
+
+	if (pContext == NULL)
+	{
+		pDevice->GetImmediateContext(reinterpret_cast<ID3D11DeviceContext**>(&pContext));
+	}
+
+	mHackerDevice = pDevice;
+	mHackerContext = pContext;
+	pDevice->SetHackerSwapChain(this);
+
+	// TODO: multisampled swap chain
+	// TODO: swap chain with more than one back buffer
+	D3D11_TEXTURE2D_DESC fake_buffer_desc;
+	std::memset(&fake_buffer_desc, 0, sizeof(D3D11_TEXTURE2D_DESC));
+	fake_buffer_desc.ArraySize = 1;
+	fake_buffer_desc.MipLevels = 1;
+	fake_buffer_desc.BindFlags = D3D11_BIND_RENDER_TARGET | D3D11_BIND_SHADER_RESOURCE;
+	fake_buffer_desc.Usage = D3D11_USAGE_DEFAULT;
+	fake_buffer_desc.SampleDesc.Count = 1;
+	fake_buffer_desc.Format = fake_swap_chain_desc->BufferDesc.Format;
+	fake_buffer_desc.MiscFlags = 0;
+	fake_buffer_desc.Width = fake_swap_chain_desc->BufferDesc.Width;
+	fake_buffer_desc.Height = fake_swap_chain_desc->BufferDesc.Height;
+	fake_buffer_desc.CPUAccessFlags = 0;
+
+	HRESULT res = pDevice->CreateTexture2D(&fake_buffer_desc, nullptr, &mFakeBackBuffer);
+
+	mWidth = new_width;
+	mHeight = new_height;
+
+	if (res != S_OK)
+	{
+		LogInfo(" *** Failed to create faked swap chain buffer! \n");
+		mFakeBackBuffer = nullptr;
+		throw;
```

Comment:
OK, took a look for our code history, but don't find any prior usage of throw for that.  Went ahead and took a look out our other uses of throw just to see if we might have some holes, but it looks mostly OK.  

However, for this scenario, it does seem like throw is the only option for error handling there.  For standard C++ constructors, there is no error return, and the error handling is just to throw and exception.

In this case, while we could continue running in the scenarios where the texture or 2nd bb failed, it doesn't seem worth that much effort.  (to separate out that into a different initializer function)

So I'd say throw is probably OK here as the way to handle possible DX11 errors.  If this proves to be a problem, we can change it later, but for now, I'd say go with the throw.  

<br>
Add a flush for the log file before the throw, so that the log is definitive on what happened.  

And probably add a bit more info to the LogInfo to explain what happened so the user knows what to try, and that the problem is related to Upscaling.